// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.confidential_match.lookup_server.proto_backend;

option java_package = "com.google.cm.lookupserver.backend";
option java_outer_classname = "ExporterDataRowProto";

// Represents a row of match data found in object storage.
//
// This proto must match against the `MatchDataRow` proto defined within
// Match Data Exporter in order for data to be deserialized correctly.
message ExporterDataRow {
  // The key associated with the data used for matching.
  bytes data_key = 1;

  // All match data associated with the key.
  repeated ExporterAssociatedData associated_data = 2;
}

// Represents all data associated with a single match record.
message ExporterAssociatedData {
  // Metadata associated with the row.
  // This can include type, user IDs, or any arbitrary information.
  message KeyValue {
    // The key associated with the key-value pair.
    string key = 1;

    // The value associated with the key-value pair.
    oneof value {
      string string_value = 2;
      int64 int_value = 3;
      double double_value = 4;
      bool bool_value = 5;
      bytes bytes_value = 6;
    }
  }
  repeated KeyValue metadata = 1;
}
